### ISSUE TITLE
Inconsistency in naming of speeches by as.speeches() when input is subcorpus

### ISSUE THREAD

**Issue by ChristophLeonhardt** [PolMine/polmineR#219]
I noticed an interesting inconsistency in the naming of speeches by `as.speeches()`. If `as.speeches()` is used on a subcorpus, depending on the s-attribute used to create the subcorpus, the naming of the subcorpora in the speech bundle changes.

For corpora, partitions and most subcorpora, the naming pattern of the objects within the speech bundle is "SPEAKER-NAME_DATE_SPEECH-INDEX".

This works in the three following cases:

### Corpus

```
speeches_in_corpus <- corpus("GERMAPARL") %>%
  as.speeches(s_attribute_name = "speaker", s_attribute_date = "date")
```

### Subcorpus based on structural attributes other than "date"

```
speeches_in_subcorpus <- corpus("GERMAPARL") %>%
  subset(year == 2000) %>%
  as.speeches(s_attribute_name = "speaker", s_attribute_date = "date")
```

### Partition

```
speeches_in_partition <- "GERMAPARL" %>%
  partition(year = 2000) %>%
  as.speeches(s_attribute_name = "speaker", s_attribute_date = "date")
```

### Subcorpus based on structural attribute "date"

When the subcorpus is based on the date itself, this naming scheme changes to "SPEAKER-NAME__SPEECH-INDEX". The date is not included here. 

I sampled the day here to check if this occurs regardless of the specific date. This seems to be the case.

```
sample_day <- sample(s_attributes("GERMAPARL", "date"), 1)

speeches_in_subcorpus_with_date <- corpus("GERMAPARL") %>%
  subset(date == sample_day) %>%
  as.speeches(s_attribute_name = "speaker", s_attribute_date = "date")
```

### Partition based on structural attribute "date"

This change in the naming convention does not seem to happen for partitions however.

```
sample_day <- sample(s_attributes("GERMAPARL", "date"), 1)

speeches_in_partition_with_date <- "GERMAPARL" %>%
  partition(date = sample_day) %>%
  as.speeches(s_attribute_name = "speaker", s_attribute_date = "date")
```

So, in conclusion the `as.speeches()` method seems to deviate from the naming scheme used for other objects when the input is a subcorpus based on "date". I think that for consistency, it would be desirable to keep including the date in the name for all outputs of `as.speeches`.

**Comment by ablaette** [comment]
Fixed, this is the code I used:

```r
corpus("GERMAPARLMINI") %>%
  subset(date == "2009-10-28") %>%
  as.speeches(s_attribute_name = "speaker", s_attribute_date = "date", verbose = FALSE) %>%
  names()
```
